docs/tutorial/csrman_design.md

```diff
@@ -1,21 +1,119 @@
 
-# Motivation for CSR Manager
+# CSR Manager Interfaces
 
+There are two sides to the CSR interfaces. First is the **(3) Snitch core interface** that handles transactions between the CPU core and the CSR manager. You control these interfaces using RISCV CSR instructions. This will be discussed in the [Programming Your Accelerator](./programming.md) section.
+
+There are two sets of channels, the request channel and the response channel. The request channel is where you specify a write or read request to the CSR manager. The response channels are the responses of the CSR manager to the Snitch core. Both channels use the decoupled interface (valid-ready protocol). Each request and response signal channels are tagged with `req` and `rsp` in the table below:
+
+|  signal name         |  channel type  | description                                  |
+| :------------------: | :------------: | :------------------------------------------: |
+| csr_req_addr_i[31:0] | req            |  Register address                            |
+| csr_req_data_i[31:0] | req            |  Data to write unto the specified address    |
+| csr_req_write_i      | req            |  Write/read signal: 1 - write, 0 read        |
+| csr_req_valid_i      | req            |  Transaction is valid                        |
+| csr_req_ready_o      | req            |  CSR manager is ready to accept transactions |
+| csr_rsp_data_o[31:0] | rsp            |  Data in response to reads                   |
+| csr_rsp_valid_o      | rsp            |  Response data is valid                      |
+| csr_rsp_ready_i      | rsp            |  Core side is ready to accept transactions   |
+
+
+The second side is the **(4) accelerator datapath interface**. The CSR manager sends the configurations of the read-write (RW) registers through a decoupled interface. However, the accelerator may have read-only (RO) registers that are meant for status monitoring. These are directly wired from the accelerator towards the CSR manager but without the need for a decoupled interface. When a user reads one of the RO registers, the CSR manager directly transmits the wired RO signals to the response ports. The table below describes these signals:
+
+|  signal name                      | description                                         |
+| :-------------------------------: | :-------------------------------------------------: |
+| csr_reg_rw_set_o [NumRwCsr][31:0] |  Packed RW configurations to send to accelerator    |
+| csr_reg_set_valid_o               |  Configurations are valid                           |
+| csr_reg_set_ready_i               |  Accelerator side is ready to accept configurations |
+| csr_reg_ro_set_i [NumRoCsr][31:0] |  Packed RO register data                            |
+
+The `NumRwCsr` and `NumRoCsr` pertain to the number of RW and RO registers, respectively.
```

Comment:
Can you already mention the  `NumRwCsr` and `NumRoCsr` are configurable design time parameters?